lib/hooks/useEvent.lua

```diff
@@ -92,10 +153,11 @@ local function useEvent(instance, event): () -> (number, ...any)
 		storage.queue = queue
 		storage.event = event
 
-		local connection = event:Connect(function(...)
+		local connection = connect(instance, function(...)
 			queue:pushBack(table.pack(...))
-		end)
+		end, event)
 
+		validateConnection(connection)
```

Comment:
I don't think we need an extra assertion when `disconnect` already checks whether it is a function, rbxscriptconnection.. etc before disconnecting it.